### PR TITLE
Add input/output filter + weaken WPlayerInvPanel's components' protection

### PR DESCRIPTION
**Pull request by Luligabi1**
- Deprecates `WItemSlot`'s filter system in favor of input and output filters;
    - The test mod is updated to test this.
- Weaken `WPlayerInvPanel`'s components (`inv`, `hotbar` and `label`) from `private` to `protected;
- Bump some dependencies, as the test mod wouldn't launch without it.

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -3,7 +3,7 @@ org.gradle.jvmargs=-Xmx1G
 
 # Fabric Properties
 	# check these on https://fabricmc.net/develop
-	minecraft_version=1.20
+	minecraft_version=1.20.1
 	yarn_mappings=1.20+build.1
```
Comment (Juuxel) [inline gradle.properties:7]: If you bump the game version, you should also update Yarn to the latest version for that MC version.

File: src/main/java/io/github/cottonmc/cotton/gui/widget/WItemSlot.java [hunk 2]
```diff
@@ -400,29 +405,73 @@ public void setBackgroundPainter(@Nullable BackgroundPainter painter) {
...
+	}
+
 	/**
 	 * Gets the item filter of this item slot.
 	 *
 	 * @return the item filter
+	 * @deprecated Replaced by {@link #getInputFilter()}
 	 * @since 2.0.0
 	 */
 	public Predicate<ItemStack> getFilter() {
```
Comment (Juuxel) [inline src/main/java/io/github/cottonmc/cotton/gui/widget/WItemSlot.java:461]: Deprecated members should have `@Deprecated`.

File: src/main/java/io/github/cottonmc/cotton/gui/widget/WPlayerInvPanel.java [hunk 3]
```diff
@@ -14,10 +14,10 @@
  * A player inventory widget that has a visually separate hotbar.
  */
 public class WPlayerInvPanel extends WPlainPanel {
-	private final WItemSlot inv;
-	private final WItemSlot hotbar;
+	protected final WItemSlot inv;
```
Comment (Juuxel) [inline src/main/java/io/github/cottonmc/cotton/gui/widget/WPlayerInvPanel.java:17]: 1. Needlessly abbreviated names like `inv` shouldn't be exposed as APIs, it needs a human-readable name
2. Public APIs should have javadoc

File: src/testMod/java/io/github/cottonmc/test/TestDescription.java [hunk 4]
```diff
@@ -40,32 +40,34 @@ public TestDescription(ScreenHandlerType<?> type, int syncId, PlayerInventory pl
 		root.add(buttonA, 0, 3, 4, 1);
 
 		WButton buttonB = new WButton(Text.literal("Show Warnings"));
-		buttonB.setOnClick(() -> slot.setIcon(new TextureIcon(new Identifier("libgui-test", "saddle.png"))));
+		buttonB.setOnClick(() -> {
+			slot.setIcon(new TextureIcon(new Identifier("libgui-test", "saddle.png")));
+		});
+
```
Comment (Juuxel) [inline src/testMod/java/io/github/cottonmc/test/TestDescription.java:46]: Don't do formatting changes like this.

**Review by Juuxel — CHANGES_REQUESTED** [review]

File: gradle.properties
Comment (Juuxel) [inline gradle.properties:1]: This file should use the original formatting (however ugly it is)

File: src/main/java/io/github/cottonmc/cotton/gui/ValidatedSlot.java [hunk 5]
```diff
@@ -15,7 +15,7 @@
 
 public class ValidatedSlot extends Slot {
 	/**
-	 * The default {@linkplain #setFilter(Predicate) item filter} that allows all items.
+	 * The default {@linkplain #setInputFilter(Predicate)  item filter} that allows all items.
```
Comment (Juuxel) [inline src/main/java/io/github/cottonmc/cotton/gui/ValidatedSlot.java:18]:
```suggestion
	 * The default {@linkplain #setInputFilter(Predicate) item filter} that allows all items.
```

File: src/testMod/java/io/github/cottonmc/test/TestDescription.java [hunk 6]
```diff
@@ -44,28 +44,22 @@ public TestDescription(ScreenHandlerType<?> type, int syncId, PlayerInventory pl
 
 		root.add(buttonB, 5, 3, 4, 1);
 		TextureIcon testIcon = new TextureIcon(new Texture(new Identifier("libgui-test", "icon.png")));
+
+
 		root.add(new WButton(testIcon, Text.literal("Button C")), 0, 5, 4, 1);
```
Comment (Juuxel) [inline src/testMod/java/io/github/cottonmc/test/TestDescription.java:49]:
```suggestion
		TextureIcon testIcon = new TextureIcon(new Texture(new Identifier("libgui-test", "icon.png")));
		root.add(new WButton(testIcon, Text.literal("Button C")), 0, 5, 4, 1);
```

**Review by Juuxel — CHANGES_REQUESTED** [review]
LGTM with these small changes

File: gradle.properties
Comment (Juuxel) [reply gradle.properties:1]: It still has formatting changes but I guess I can fix those when merging

**Review by Juuxel — APPROVED** [review]

File: gradle.properties
Comment (Luligabi1) [reply gradle.properties:1]: oops, I ended up picking it from my previous-previous commit instead of the original. Sorry!

**Pull request merged**